sdmetrics/reports/multi_table/_properties/relationship_validity.py

```diff
@@ -0,0 +1,134 @@
+import numpy as np
+import pandas as pd
+import plotly.express as px
+
+from sdmetrics.column_pairs.statistical import CardinalityBoundaryAdherence, ReferentialIntegrity
+from sdmetrics.reports.multi_table._properties.base import BaseMultiTableProperty
+from sdmetrics.reports.utils import PlotConfig
+
+
+class RelationshipValidity(BaseMultiTableProperty):
+    """``Relationship Validity`` property.
+
+    This property measures the validity of the relationship
+    from the primary key and the foreign key perspective.
+
+    """
+
+    _num_iteration_case = 'relationship'
+
+    def _generate_details(self, real_data, synthetic_data, metadata, progress_bar=None):
+        """Generate the _details dataframe for the relationship validity property.
+
+        Args:
+            real_data (dict[str, pandas.DataFrame]):
+                The tables from the real dataset, passed as a dictionary of
+                table names and pandas.DataFrames.
+            synthetic_data (dict[str, pandas.DataFrame]):
+                The tables from the synthetic dataset, passed as a dictionary of
+                table names and pandas.DataFrames.
+            metadata (dict):
+                The metadata, which contains each column's data type as well as relationships.
+            progress_bar (tqdm.tqdm or None):
+                The progress bar object. Defaults to ``None``.
+
+        Returns:
+            float:
+                The average score for the property for all the individual metric scores computed.
+        """
+        child_tables, parent_tables = [], []
+        primary_key, foreign_key = [], []
+        metric_names, scores, error_messages = [], [], []
+        metrics = [ReferentialIntegrity, CardinalityBoundaryAdherence]
+        for relation in metadata.get('relationships', []):
+            real_columns = self._extract_tuple(real_data, relation)
+            synthetic_columns = self._extract_tuple(synthetic_data, relation)
+            for idx, metric in enumerate(metrics):
+                try:
+                    relation_score = metric.compute(
+                        real_columns,
+                        synthetic_columns,
+                    )
+                    error_message = None
+                except Exception as e:
+                    relation_score = np.nan
+                    error_message = f'{type(e).__name__}: {e}'
+                finally:
+                    if progress_bar is not None and idx % 2 == 0:
+                        progress_bar.update()
```

Comment:
Instead of `idx %2 == 0` could we do `idx % len(metrics) == 0? Or maybe just remove the update from this for loop entirely? I think rather than running this as a `finally` we can move it outside of the metric `for` loop and just update for each relationship and then also change `for idx, metric in enumerate(metrics)` to just `for metric in metrics`.